Sync nes and hapi-auth-basic plugin definitions to Hapi 17.

saboya opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

This might change based on https://github.com/DefinitelyTyped/DefinitelyTyped/issues/23562 resolution, but it's better to merge this anyway since the types are Hapi >= 17 exclusive.

DanielRosenwasser commented on types/nes/index.d.ts:
```diff
@@ -23,114 +23,127 @@
-
-    interface ClientSubscribeFlags extends NesClient.ClientSubscribeFlags {}
+	interface SocketAuthObject {
+		isAuthenticated: boolean;
```
You've converted the file to mix tabs and spaces. Can you undo this please?

SimonSchick commented on types/nes/index.d.ts:
```diff
@@ -23,114 +23,127 @@
+
+	interface ServerSubscriptionOptionsFilterOptions {
+		socket: Socket;
+		credentials?: any;
```
See above.

SimonSchick commented on types/nes/index.d.ts:
```diff
@@ -23,114 +23,127 @@
-    interface ClientSubscribeFlags extends NesClient.ClientSubscribeFlags {}
+	interface SocketAuthObject {
+		isAuthenticated: boolean;
+		credentials: any;
```
You might want to make those interfaces so users can define these as they wish.

SimonSchick commented on types/nes/index.d.ts:
```diff
@@ -23,114 +23,127 @@
+	}
+
+	interface ServerBroadcastOptions {
+		user: any
```
See above.

SimonSchick commented on types/nes/index.d.ts:
```diff
@@ -23,114 +23,127 @@
+		index?: boolean;
+	}
+
+	export type ServerOnSubscribeWithParams = (socket: Socket, path: string, params: any) => Promise<any>;
```
Ca we narrow `any`?

SimonSchick commented on types/nes/test/broadcast-server.ts:
```diff
@@ -1,14 +1,14 @@
 import Nes = require('nes');
 
-var server = new Hapi.Server();
+const server: Server = new Server();
```
Redundant typings imo.

SimonSchick commented on types/nes/test/client-require.ts:
```diff
@@ -1,12 +1,12 @@
 // from https://github.com/hapijs/nes/#browser-client
-// When you require('nes') it loads the full module and adds a lot of extra code
+// When you require("nes") it loads the full module and adds a lot of extra code
```
Why the random quoting change?

SimonSchick commented on types/nes/test/nes-tests.ts:
```diff
@@ -1,56 +1,56 @@
+var server: Server = new Server();
 
 server.register(Nes).then(() => {
-    // No longer need to cast to Nes.Server as Hapi.Server has been modified directly.
```
Pls don't randomly change whitespaces.

saboya commented:
@DanielRosenwasser @SimonSchick My apologies for the whitespace mess, my editor didn't handle a project this big well, so I deviated from my current workflow and thought I was actually "fixing" the whitespacing / quoting. I corrected everything and rebased, along with the unnecessary typings and some unnecessary imports as well.

As for the other issues, I'm gonna take a look later. My rationale behind this update was just to keep it up to date with the new Hapi bindings, so I didn't have any improvements in mind, so that definitely got overlooked.

Thanks for the input.

DanielRosenwasser commented:
Thanks!
